source/agora/common/crypto/Schnorr.d

```diff
@@ -399,6 +399,69 @@ nothrow @nogc @safe unittest
     assert(stolen_key != kp.v);
 }
 
+/// Single-signer as part of multi-sig
+public Signature multiSigSign (const ref Pair R,
+    const ref Scalar v, const ref Scalar c) nothrow @nogc @safe
+{
+    Scalar s = R.v + (v * c);
+    return Sig(R.V, s).toBlob();
+}
+
+/// multi-sig verify
+public bool multiSigVerify (const ref Signature signature,
+    const ref Point sumV, const ref Scalar c) nothrow @nogc @safe
+{
+    Sig sigRs = Sig.fromBlob(signature);
+    return sigRs.s.toPoint() == sigRs.R + (sumV * c);
+}
+
+public Signature multiSigCombine (const ref Signature[] signatures) nothrow @nogc @safe
+{
+    Point sum_R;
+    Scalar sum_s;
+    signatures.each!(sig => {
+        Sig sigRs = Sig.fromBlob(sig);
+        sum_R = sum_R == Point.init ? sigRs.R : (sum_R + sigRs.R);
+        sum_s = sum_s == Scalar.init ? sigRs.s : (sum_s + sigRs.s);
+    } ());
```

Comment:
I think there's something to `sum` with a `seed` in Phobos.

Reply:
I updated to use Phobos sum with seed as `Point.init` and `Scalar.init`. However this did mean updating the binary functions in Point and Scalar for addition. It now treats the init values as `functional identity`.